cmd/dagger/main.go

```diff
@@ -172,6 +174,15 @@ func installGlobalFlags(flags *pflag.FlagSet) {
 	}
 }
 
+// disableFlagsInUseLine disables the automatic addition of [flags]
+// when calling UseLine.
+func disableFlagsInUseLine(cmd *cobra.Command) {
+	for _, c := range cmd.Commands() {
+		c.DisableFlagsInUseLine = true
+		disableFlagsInUseLine(c)
```

Comment:
I see you didn't use [my sugestion](https://github.com/dagger/dagger/pull/7107#discussion_r1569024798) of the custom useLine:
```go
c.Use = useLine(c)
```

```go
// custom useLine function that prints options instead of flags.
func useLine(c *cobra.Command) string {
	return strings.Replace(c.UseLine(), "[flags]", "[options]")
}
```
It would replace the need to manually add `[options]` to a lot of commands, although I'll also accept with them manually added due to explicitness.

Reply:
I tried, but there were two reasons for that:
1. the useLine as suggested broke the parsing of the commands, as the c.UseLine string starts with the program name and the command, whereas the 'Use' field of the command is used for the command parsing.

With the useLine(c) function:
```shell
./dagger listen
Error: unknown command "listen" for "dagger"
Run 'dagger --help' for usage.
```

We could strip the "dagger " part of the `UseLine()` string, but is it really worth it as the explicitness could be more readable ?

Working implementation:
```go
// useLine replaces "[flags]" with "[options]" on the usage
// it keeps the same c.Use format (without CLI name), as it is being used for parsing
func useLine(c *cobra.Command) string {
	usageLine := c.UseLine()
	modifiedUsageLine := strings.Replace(usageLine, "[flags]", "[options]", -1)
	
	// Remove the "dagger " prefix by slicing the string from the 7th index onwards.
	// breaks command parsing otherwise
	if strings.HasPrefix(modifiedUsageLine, "dagger ") {
		return modifiedUsageLine[7:]
	}
	return modifiedUsageLine
}
```

2.  Explicitly specifying the `[options]` ensures the position of the appended `[flags]`, where for example, in our current CLI, the position of `[flags]` is not coherent across commands:
```shell
dagger install --help
Add a Dagger module as a dependency of a local module.

Usage:
  dagger install [flags] MODULE
```

```shell
dagger gen --help
Generate CLI reference documentation in the given file path.

Usage:
  dagger gen FILE [flags]
```
But, this could also potentially conflict with https://github.com/dagger/dagger/pull/7143#discussion_r1577045429, so no strong opinion on that

Reply:
This isn't making any sense. Which implementation of UseLine are you referring to? I put it on the first comment here.

Reply:
I linked to the original suggestion, have you taken it into consideration? What you're describing doesn't reflect what I said.

Reply:
Oh wait, yeah I see it now. I'll get back to you on that (I'm on phone), in the meantime disregard my feedback here.